src/Assert.php

```diff
@@ -1688,6 +1855,17 @@ protected static function strlen($value)
         return \mb_strlen($value, $encoding);
     }
 
+    private static function throwException($message, $exception = null)
+    {
+        if (!is_null($exception)) {
+            if ($exception instanceof Exception) {
```

Comment:
Besides my other comment, there should be a check for both `Exception` and `Throwable` here.

Even though it doesn't exist in lower PHP versions, there won't be an error from trying to catch a non existent class.